File: frontend/src/features/externalCourses/externalCourseModels.ts
export type ExternalContentStatus =
  | 'TaskCreated'
  | 'ReviewRequired'
  | 'NotVisible'

export interface CourseSubscription {
  id: string
  moduleId: string
  courseName: string
  providerKey: string
  externalCourseId: string
  lastScanStatus: string
  lastSuccessfulScanAtUtc: string | null
}

export interface ExternalCourseContent {
  id: string
  providerContentId: string
  title: string
  description: string | null
  sourceUrl: string
  dueDateUtc: string | null
  status: ExternalContentStatus
  reviewReason: string | null
  taskId: string | null
}

export interface CourseScanSummary {
  status: string
  newContentCount: number
  changedContentCount: number
  reviewRequiredCount: number
  notVisibleCount: number
  newTaskEligibleCount: number
}

export interface RegisterCourseRequest {
  courseUrl: string
}

File: backend/src/Api/ExternalCourses/ExternalCourseModels.cs
using System.ComponentModel.DataAnnotations;

namespace StudyOrganizer.Api.ExternalCourses;

public sealed class RegisterCourseSubscriptionRequest
{
    [Required]
    [StringLength(2048)]
    [Url]
    public string CourseUrl { get; init; } = string.Empty;
}

public sealed record CourseSubscriptionResponse(
    Guid Id,
    Guid ModuleId,
    string CourseName,
    string ProviderKey,
    string ExternalCourseId,
    string LastScanStatus,
    DateTimeOffset? LastSuccessfulScanAtUtc);

public sealed record ExternalCourseContentResponse(
    Guid Id,
    string ProviderContentId,
    string Title,
    string? Description,
    string SourceUrl,
    DateTimeOffset? DueDateUtc,
    string Status,
    string? ReviewReason,
    Guid? TaskId);

public sealed record CourseScanResponse(
    string Status,
    int NewContentCount,
    int ChangedContentCount,
    int ReviewRequiredCount,
    int NotVisibleCount,
    int NewTaskEligibleCount);
